sensu/gohandler.go

```diff
@@ -6,16 +6,16 @@ import (
 	"os"
 )
 
-type GoHandler struct {
-	GoPlugin
+type goHandler struct {
+	basePlugin
 	validationFunction func(event *types.Event) error
 	executeFunction    func(event *types.Event) error
 }
 
 func NewGoHandler(config *PluginConfig, options []*PluginConfigOption,
-	validationFunction func(event *types.Event) error, executeFunction func(event *types.Event) error) *GoHandler {
-	goHandler := &GoHandler{
-		GoPlugin: GoPlugin{
+	validationFunction func(event *types.Event) error, executeFunction func(event *types.Event) error) GoPlugin {
```

Comment:
This function should return a `goHandler`. Since a `goHandler` should satisfy the interface of a `GoPlugin`, the type assertions used later in the code are not necessary. 

See https://medium.com/rungo/interfaces-in-go-ab1601159b3a for a nice, simple example on satisfying interfaces.

Reply:
Maybe it's my Java background but I believe `NewGoHandler` and `NewGoMutator` should return the `GoPlugin` interface instead of a concrete type. The idea is to only expose only the `Execute` method to the client code. Since this is a library the only code invoking this function is the unit test. In this case I only use the type assertion as way to perform some validation and to override the `os.Exit` method's behavior so the test does not terminate. You can see an example of how this is invoked in an actual handler at https://github.com/sensu-skunkworks/sensu-elasticsearch-handler/blob/setup-project/main.go#L75. Does that make sense?

Reply:
@nikkiki - I realized you might not have access to the previous repo, you can look at this snippet from the ServiceNow handler:

https://github.com/sensu/sensu-servicenow-handler/blob/use-templates/main.go#L108

Reply:
No problem, thanks for the example! If you returned the concrete type from the `New...` function, the resulting type should still be able to invoke `Execute()` since it satisfies the `GoPlugin` interface. Let me get a second opinion on this!

Reply:
@fguimond, interfaces work a bit differently in Go and it does indeed surprise many people with Java backgrounds. I agree with Nikki here, and don't see value in breaking the API at this time.

This document has some good practices for Go development that you might find interesting, including some advice about interfaces. https://github.com/golang/go/wiki/CodeReviewComments#interfaces

Reply:
Thank you for the explanation @nikkiki and @echlebek! It is indeed a different philosophy, I'll definitely read more on interfaces.

The changes were applied. I had to make the GoHandler and GoMutator struct "public" (with leading upper-case) so they can be used in other packages.